src/node_quic_util-inl.h

```diff
@@ -234,6 +188,101 @@ const char* QuicError::GetFamilyName() {
   }
 }
 
+const ngtcp2_cid* QuicPreferredAddress::cid() const {
+  return &paddr_->cid;
+}
+
+const uint8_t* QuicPreferredAddress::StatelessResetToken() const {
+  return paddr_->stateless_reset_token;
+}
+
+const sockaddr_in6* QuicPreferredAddress::PreferredIPv6Address() const {
+  return reinterpret_cast<const sockaddr_in6*>(&paddr_->ipv6_addr);
+}
+const sockaddr_in* QuicPreferredAddress::PreferredIPv4Address() const {
+  return reinterpret_cast<const sockaddr_in*>(&paddr_->ipv4_addr);
+}
+
+const int16_t QuicPreferredAddress::PreferredIPv6Port() const {
+  return paddr_->ipv6_port;
+}
+const int16_t QuicPreferredAddress::PreferredIPv4Port() const {
+  return paddr_->ipv4_port;
+}
+
+bool QuicPreferredAddress::Use(int family) const {
+  uv_getaddrinfo_t req;
+
+  if (!ResolvePreferredAddress(family, &req) ||
+      req.addrinfo == nullptr) {
```

Comment:
Should `uv_freeaddrinfo()` be called in the second case?

Reply:
No, if `ResolvePreferredAddress()` returns false, then there's no need to call `uv_freeaddrinfo`

Reply:
Yeah, I know, but in the `req.addrinfo == nullptr` case that isn’t quite as obvious?

Reply:
If req.addrinfo is nullptr, there's nothing to free.